src/modules/haxe.rs

```diff
@@ -0,0 +1,252 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::haxe::HaxeConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+use serde_json as json;
+
+use regex::Regex;
+const HAXE_VERSION_PATTERN: &str =
+    "(?P<version>(?:\\d+\\.\\d+\\.\\d+(?:-(?:alpha|beta|pre|rc)[\\d\\.+]+)?)|[[:xdigit:]]+)";
+
+/// Creates a module with the current Haxe version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("haxe");
+    let config = HaxeConfig::try_load(module.config);
+
+    let is_haxe_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_haxe_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let haxe_version = get_haxe_version(context)?;
+                    VersionFormatter::format_module_version(
+                        module.get_name(),
+                        &haxe_version,
+                        config.version_format,
+                    )
+                    .map(Ok)
+                }
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `haxe`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_haxe_version(context: &Context) -> Option<String> {
+    let file_contents = context.read_file_from_pwd(".haxerc");
+    if file_contents.is_some() {
+        let raw_json = match file_contents {
+            Some(val) => val,
+            None => panic!("failed to read contents of .haxerc!"),
+        };
+        let haxerc_json: json::Value = json::from_str(&raw_json).ok()?;
+        let raw_version = haxerc_json.get("version")?.as_str()?;
```

Comment:
Does the `haxerc` always contain `version`? If not, it should fall back to `haxe --version`

Reply:
`.haxerc` always contains a version field.
however not all projects use lix package manager, so `.haxerc` is not present for all Haxe projects.

after checking lix sources there are multiple different values that `version` field can represent. it can either be a semver version, a git hash of a nightly build, fixed names like `stable` or `latest` or even a path to a local build of Haxe compiler. given these different possible values it doesn't make a lot of sense to use a regex to validate it's `version` field contents.

also `haxe --version` only prints a version string, so I might drop regex check entirely. unless there is a good reason to keep validating version information. 

regardless I will add a fallback to `haxe --version` in case of json parser issues.

Reply:
> it can either be a semver version, a git hash of a nightly build, fixed names like stable or latest or even a path to a local build of Haxe compiler.

In that case, I think there should be some verification that the field contents are version-like to avoid displaying long paths in the prompt.

Reply:
I've added code to detect a path in a `.haxerc` version field and it will call the exectuable instead to get it's actual version number.